engine/resources/pippet_run.go

```diff
@@ -0,0 +1,160 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+)
+
+// PippetRunner is the interface to the wrapped "puppet yamlresource receive"
+// process that keeps running alongside mgmt. Individual resources are expected
+// to register themselves, and can use the Apply function from that point on.
+type PippetRunner interface {
```

Comment:
I'm not sure why you have an interface type. Interfaces are useful (in general) when more than one thing implements them, or when you plan a future second implementation. AIUI there's only one here...

Reply:
It will make testing much easier :)

Reply:
If so, then can you write a small stub test to show how it's useful/easier that way please?